### ISSUE TITLE
R_X86_64_IRELATIVE ELF relocations not handled

### ISSUE THREAD

**Issue by Dave401** [NationalSecurityAgency/ghidra#1189]
**Describe the bug**
The ELF relocation type R_X86_64_IRELATIVE is explicitly not handled by Ghidra. In the X86_64_ElfRelocationHandler file, the case for this type has the comment: "NOTE: We don't support this since the code actually uses a function to compute the relocation value (i.e., indirect)". However, both IDA and objdump seem to handle this type quite well, at least in the example I am looking at. 

Below are screenshots of output from IDA, objdump, and Ghidra for the same segment of the .got.plt segment of the libc.so.6 binary from Ubuntu 18.04 (x64). 

Here is the IDA output:

![IDA](https://user-images.githubusercontent.com/56848547/67813696-91c2e800-fa78-11e9-8018-b5de048abcfc.jpg)

Here is the objdump output:

![objdump](https://user-images.githubusercontent.com/56848547/67813735-aa330280-fa78-11e9-9a70-2b44a7abe1b0.jpg)

Here is the Ghidra output:

![Ghidra](https://user-images.githubusercontent.com/56848547/67813739-adc68980-fa78-11e9-9411-f1f75bdcfa9a.jpg)

Note that Ghidra uses a different image base than the other programs, so the addresses are 0x3eb018-0x3eb038 for IDA and objdump and 0x4eb018-0x4eb038. You can see that the last value at 0x3eb038/0x4eb038 has a relocation type R_X86_64_JUMP_SLOT and is handled well in Ghidra. However, the previous 4 values have the R_X86_64_IRELATIVE relocation types and so Ghidra just shows unhelpful values. In the IDA output you can see they are set to the addresses of functions in the binary which appears to be the expected behavior. In the objdump output, the offset shown is the offset to the associated function from the image load address (e.g., the offset shown for the relocation at address 0x3eb018 is 0x9ec70, which is the offset to memmove, matching the IDA output).

Since this binary's .got.plt section uses's this relocation type a lot, most of the section is unreadable in Ghidra.

**Comment by mumbel** [comment]
so is it as simple as doing:
```
case X86_64_ElfRelocationConstants.R_X86_64_IRELATIVE:
    value = addend + elfRelocationContext.getImageBAseWordAdjustmentOffset();
    memory.setLong(relocationAddress, value);
```

**Comment by LloydLabs** [comment]
Did this issue ever get addressed? I'm getting this issue in 9.2.2 in the following environment:

```
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19042 N/A Build 19042

openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)
```

**Comment by ghidra1** [comment]
Sorry, this appears to have slipped under the radar.  I will try to work it in sooner than later.

**Comment by ghidra1** [comment]
Corrected for Ghidra 9.2.3

**Comment by LloydLabs** [comment]
@ghidra1 amazing, thank you!
